### PR TITLE
feat: camera edits & other miscellaneous tasks

### LINKED ISSUE #138: Create player look at
Sometimes the camera zooms in when does a block of 1 block between the character and the camera (on the same plane). This causes unnecessary zoom as you can still see the character.

Lets use a `PlayerLookAt` on the character's head so that the camera zoom only happens when the top of the character is blocked.

### LINKED ISSUE #152: first click after exiting the menu is treated as an ingame click
if reopening the menu immediately without clicking, the next click will close the menu automatically and pull a block if applicable

edit: i tried unpausing the game with a keyboard button and it doesn't have this issue, so it is something to do with mouse click?

### PR DESCRIPTION
**Pull request by Rinder5**
fixes #138 
fixes #152

feat: camera clamps less (allows more freeform camera rotation)
feat: camera checks higher for body collision (prevents awkward camera zooms)
feat: camera rotates back on unpause (reduces player disorientation)
feat: camera resets to starting position on reload and change scene (i call reset camera quite a lot but it doesn't seem to work correctly without it)
fix: buttons don't click twice
feat: stage select now looks cooler (images are temporary and hackily done. Please touch up)
fix: player does not fall through floors on respawning
feat: placeholder for collectibles ui (not implemented yet)
chore: collectibles use trigger instead of collision

### CODE REVIEW

File: Assets/Scripts/CameraController.cs [hunk 1]
```diff
@@ -252,4 +260,17 @@ public void SetIdle(bool idle)
     {
         _idle = idle;
     }
+
+    public void ResetCameraAngle()
+    {
+        float yDiff = transform.eulerAngles.y - StartingYRotation;
+        RotateX(-yDiff);
+        RotateX(-currentAutoRotate);
```
Comment (nicholaschuayunzhi) [inline Assets/Scripts/CameraController.cs]: Seems like the starting rotation is fixed. So for some stages there will the camera will reset in a non-scenic view. If we have time i will work on that.

Comment (nicholaschuayunzhi) [reply Assets/Scripts/CameraController.cs]: Or maybe it should rotate according to the face direction of the character?

Comment (Rinder5) [reply Assets/Scripts/CameraController.cs]: the camera will rotate to face the level. Don't know what you mean by scenic, but it's at least practical maybe?

Comment (Rinder5) [reply Assets/Scripts/CameraController.cs]: btw, I notice how stupid this wording looks X(Y), gonna change it to make it more correct

File: Assets/Scripts/CollectiblesScript.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	void Start ()
+    {
+		
+	}
+	
+	void Update ()
+    {
+		
+	}
```
Comment (nicholaschuayunzhi) [inline Assets/Scripts/CollectiblesScript.cs]: could you fix the formatting here

File: Assets/Scripts/CollectiblesScript.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+	void Start ()
+    {
+		
+	}
+	
+	void Update ()
+    {
+        transform.RotateAround(transform.position, Vector3.up, rotateSpeed);
+	}
```
Comment (nicholaschuayunzhi) [inline Assets/Scripts/CollectiblesScript.cs]: Could you fix the formatting here?

File: Assets/Scripts/CollectiblesScript.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    private void OnTriggerEnter(Collision collision)
+    {
+        Destroy(this.gameObject);
+    }
```
Comment (nicholaschuayunzhi) [inline Assets/Scripts/CollectiblesScript.cs]: Does this work? `OnTriggerEnter` should take in a `Collider` https://docs.unity3d.com/ScriptReference/Collider.OnTriggerEnter.html

**Review by nicholaschuayunzhi — COMMENTED** [review]
Thanks for your hard work. Will merge this and continue work on the collectibles.

**Pull request merged**